src/state/user/actions.js

```diff
@@ -35,22 +43,33 @@ export const startSetUserMocs = payload => (dispatch) => {
     if (!snapshot.exists()) {
       console.log('no mocs for this user');
       dispatch(setMOCs([]));
-      return; 
+      return;
     }
-    const mocIds = Object.keys(snapshot.val());
-    Promise.all(mocIds.map((id) => {
-      return firebasedb.ref(`mocData/${id}`).once('value').then((snapshot) => {
-        let moc = snapshot.val();
-        let mocData = {
-          govtrack_id: moc.govtrack_id,
-          member_name: moc.displayName,
+    const govtrackIds = map(filter(snapshot.val(), 'govtrack_id'), 'govtrack_id');
+    const thpIds = map(filter(snapshot.val(), 'thp_id'), 'thp_id');
+    const federalLookups = map(govtrackIds, id => firebasedb.ref(`${MOC_DATA_ENDPOINT}/${id}`).once('value'));
+    const stateLooksUps = map(thpIds, id => firebasedb.ref(`${STATE_LEG_DATA_ENDPOINT}/${id}`).once('value'));
+
+    Promise.all([...federalLookups, ...stateLooksUps]).then(returned => returned.reduce((acc, mocSnap) => {
+      if (mocSnap.exists()) {
+        const moc = mocSnap.val();
+        if (!moc.displayName) {
+          return acc;
         }
-        return mocData;
-      })
-    }))
+        const mocData = {
+          ...moc,
```

Comment:
@Alan-mag since we are going to need the full moc data, we might as well download it on the first lookup and store it locally.